File: eye_test_engine/test_api.js
const http = require('http');

async function test() {
    const fetch = (await import('node-fetch')).default || globalThis.fetch;
    const base = "http://localhost:5000/api/session";
    const sid = "test10";
    
    let res = await fetch(`${base}/start`, { method: 'POST', body: JSON.stringify({session_id: sid}), headers: {'Content-Type': 'application/json'}});
    console.log("start:", (await res.json()).power.right.sph);
    
    res = await fetch(`${base}/${sid}/respond`, { method: 'POST', body: JSON.stringify({intent: "Able to read with pinhole"}), headers: {'Content-Type': 'application/json'}});
    console.log("respond:", (await res.json()).power.right.sph);
    
    res = await fetch(`${base}/${sid}/sync-power`, { method: 'POST', body: JSON.stringify({right: {sph: -1.25}}), headers: {'Content-Type': 'application/json'}});
    console.log("sync-power:", await res.json());
    
    res = await fetch(`${base}/${sid}/status`);
    console.log("after sync:", (await res.json()).current_power.right.sph);
    
    res = await fetch(`${base}/${sid}/respond`, { method: 'POST', body: JSON.stringify({intent: "Blurry"}), headers: {'Content-Type': 'application/json'}});
    console.log("blurry:", (await res.json()).power.right.sph);
}

test().catch(console.error);

File: eye_test_engine/test_api.py
import requests
import time
import subprocess

# Start server
server = subprocess.Popen(["python3", "api_server.py"])
time.sleep(2)

base = "http://localhost:5000/api/session"
sid = "test4"

try:
    print("start:", requests.post(f"{base}/start", json={"session_id": sid}).json()['power']['right']['sph'])
    print("respond (pinhole):", requests.post(f"{base}/{sid}/respond", json={"intent": "Able to read with pinhole"}).json()['power']['right']['sph'])
    
    print("before sync:", requests.get(f"{base}/{sid}/status").json()['current_power']['right']['sph'])
    
    res = requests.post(f"{base}/{sid}/sync-power", json={"right": {"sph": -1.25, "cyl": 0.0, "axis": 180}})
    print("sync-power:", res.json())
    
    print("after sync:", requests.get(f"{base}/{sid}/status").json()['current_power']['right']['sph'])
    
    res = requests.post(f"{base}/{sid}/respond", json={"intent": "Blurry"})
    print("respond (blurry):", res.json()['power']['right']['sph'])

finally:
    server.terminate()
